ships/0005-local-source-upload.md

```diff
@@ -0,0 +1,250 @@
+---
+title: local-source-upload
+authors:
+  - @otaviof
+reviewers:
+  - @alicerum
+  - @adambkaplan
+approvers:
+creation-date: 2021-06-23
+last-updated: 2021-06-23
+---
+
+# Source Upload
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [docs](/docs/)
+
+## Summary
+
+Proposal to improve developer-experience by allowing to upload a local repository clone to the
+Kubernetes cluster, and use Shipwright Build Controller to produce a container-image out of it.
+
+As a developer, I can try Shipwright before opening a pull-request, and take advantage of the cluster
+resources to create the image. Effectively moving shipwright closer to the developer's inner-loop, in
+other words we are "moving left before _shipping-right_".
+
+## Motivation
+
+Shipwright Build-Controller allows teams to build container-images out of Git repositories, register
+and track their application builds in a Kubernetes native environment. This enhancement proposal
+extends this ability for developers to upload uncommitted changes and use the same machinery during
+the inner-loop as well.
+
+As a developer you can try out changes before opening a pull-request, make sure the final
+container-image will work out as expected, and effectively improve the developer experience.
+
+The approach proposed here is complementary to existing [EP][local-source-code-ep] detailing local
+source upload to a container-registry and later on self-extracting the image before the actual
+container build process starts.
+
+### Goals
+
+- Improve developer experience, local clone can produce a container-image using Shipwright machinery
+- Move Shipwright CLI closer to developer's inner-loop
+- Work as an alternative to [upload-to-container-registry][local-source-code-ep], in near future
+
+## Proposal
+
+This proposal includes changes in Shipwright components, and adds a new helper application to aid
+data upload process.
+
+### Sequence Diagram
+
+The following sequence-diagram shows the relationship between the components and explains the order
+of execution. In a nutshell, the CLI creates a `BuildRun` object, with settings to wait for data
+upload, which can only start when the Build-Controller instructs Kubernetes to create this new pod.
+The new pod, on the other hand, needs to be running and ready to receive external data.
+
+With the data upload complete, the build-pod can continue to execute the build-strategy chosen,
+producing a container-image at the end.
+
+![source-upload sequence diagram](assets/0005-source-upload-sequence-diagram.png)
```

Comment:
love the diagram :-)

a direct link for others to see it directly:  https://github.com/shipwright-io/community/blob/9589a6a3c3407375bc3c044fad1ad5ce42bc7f2d/ships/assets/0005-source-upload-sequence-diagram.png